spanner/spanner_snippets/spanner/pg_spanner_numeric_data_type.go

```diff
@@ -0,0 +1,162 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package spanner
+
+// [START spanner_postgresql_numeric_data_type]
+
+import (
+	"context"
+	"fmt"
+	"io"
+
+	"cloud.google.com/go/spanner"
+	database "cloud.google.com/go/spanner/admin/database/apiv1"
+	"google.golang.org/api/iterator"
+	adminpb "google.golang.org/genproto/googleapis/spanner/admin/database/v1"
+)
+
+// pgNumericDataType shows how to work with the PostgreSQL NUMERIC/DECIMAL data
+// type on a Spanner PostgreSQL database.
+func pgNumericDataType(w io.Writer, db string) error {
+	// db := "projects/my-project/instances/my-instance/databases/my-database"
+	ctx := context.Background()
+
+	adminClient, err := database.NewDatabaseAdminClient(ctx)
+	if err != nil {
+		return err
+	}
+	defer adminClient.Close()
+
+	// Create a table that includes a column with data type NUMERIC. As the database has been
+	// created with the PostgreSQL dialect, the data type that is used will be the PostgreSQL
+	// NUMERIC / DECIMAL data type.
+	op, err := adminClient.UpdateDatabaseDdl(ctx, &adminpb.UpdateDatabaseDdlRequest{
+		Database: db,
+		Statements: []string{
+			`CREATE TABLE Venues (
+				VenueId  bigint NOT NULL PRIMARY KEY,
+				Name     varchar(1024) NOT NULL,
+				Revenues numeric
+			)`},
+	})
+	if err != nil {
+		return err
+	}
+	if err := op.Wait(ctx); err != nil {
+		return err
+	}
+	fmt.Fprintf(w, "Created Venues table\n")
+
+	client, err := spanner.NewClient(ctx, db)
+	if err != nil {
+		return err
+	}
+	defer client.Close()
+
+	var updateCount int64
+	// Insert a Venue using DML.
+	_, err = client.ReadWriteTransaction(ctx, func(ctx context.Context, transaction *spanner.ReadWriteTransaction) error {
```

Comment:
Line 70 is also hard to parse. Suggest breaking it into multiple statements.

I liked how `spanner.Statement` was instantiated and then passed into the `func` in a previous sample.